low-level/files/opt/seagate/sspl/setup/sspl_prepare.py

```diff
@@ -88,10 +78,25 @@ def validate(self):
             "server_node>%s>network>data>private_fqdn" % machine_id)
         data_public_fqdn = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
             "server_node>%s>network>data>public_fqdn" % machine_id)
+        enclosure_id = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+            "server_node>%s>storage>enclosure_id" % machine_id)
+        primary_ip = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+            "storage_enclosure>%s>controller>primary>ip" % enclosure_id)
+        secondary_ip = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+            "storage_enclosure>%s>controller>secondary>ip" % enclosure_id)
+        cntrlr_user = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+            "storage_enclosure>%s>controller>user" % enclosure_id)
+        cntrlr_secret = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+            "storage_enclosure>%s>controller>secret" % enclosure_id)
+        cntrlr_key = Cipher.generate_key(cluster_id,
+            ServiceTypes.STORAGE_ENCLOSURE.value)
+        cntrlr_passwd = Cipher.decrypt(cntrlr_key,
```

Comment:
how this cntrlr_passwd  is different from cntrlr_secret ?

Reply:
As given in [encryption-decryption reference guide](https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/127893714/Cipher+key+generation+for+secret+password) 

Chiper key is generated using cluster_id and root_node of the `secret` key.

Encryption:
`cntrlr_secret` is generated using that Cipher key and plain password.

Decryption:
Using the same Cipher key, we decrypt the encrypted text. it gives back the plain password(`password`) used with Encryption.

Reply:
ok